Mathlib/Analysis/Calculus/LogDeriv.lean

```diff
@@ -105,3 +106,48 @@ theorem logDeriv_comp {f : 𝕜' → 𝕜'} {g : 𝕜 → 𝕜'} {x : 𝕜} (hf
     (hg : DifferentiableAt 𝕜 g x) : logDeriv (f ∘ g) x = logDeriv f (g x) * deriv g x := by
   simp only [logDeriv, Pi.div_apply, deriv_comp _ hf hg, comp_apply]
   ring
+
+lemma logDeriv_eqOn_iff {E 𝕜 : Type*} [NontriviallyNormedField E]
+    [NontriviallyNormedField 𝕜] [IsRCLikeNormedField 𝕜] [NormedAlgebra 𝕜 E] [NormedSpace ℝ 𝕜]
+    {f g : 𝕜 → E} {s : Set 𝕜} (hf : DifferentiableOn 𝕜 f s) (hg : DifferentiableOn 𝕜 g s)
+    (hs2 : IsOpen s) (hsc : Convex ℝ s) (hgn : ∀ x ∈ s, g x ≠ 0) (hfn : ∀ x ∈ s, f x ≠ 0) :
```

Comment:
Is convexity of `s` really needed? I think you can weaken this to preconnectedness, using `IsOpen.exists_is_const_of_deriv_eq_zero` in place of `Convex.is_const_of_fderivWithin_eq_zero`.